src/sinks/datadog/metrics/tests.rs

```diff
@@ -144,3 +144,32 @@ async fn smoke() {
         assert_eq!(points.get(1).unwrap().as_f64().unwrap(), 1.0);
     }
 }
+
+#[tokio::test]
+async fn real_endpoint() {
+    let config = indoc! {r#"
+        default_api_key = "${DATADOG_API_KEY}"
+        default_namespace = "fake.test.integration"
+    "#};
+    let api_key = std::env::var("CI_TEST_DATADOG_API_KEY").unwrap();
+    let config = config.replace("${DATADOG_API_KEY}", &api_key);
+    let (config, cx) = load_sink::<DatadogMetricsConfig>(config.as_str()).unwrap();
+
+    let (sink, _) = config.build(cx).await.unwrap();
+    let (batch, receiver) = BatchNotifier::new_with_receiver();
+    let events: Vec<_> = (0..10)
+        .map(|index| {
+            Event::Metric(Metric::new(
+                "counter",
+                MetricKind::Absolute,
+                MetricValue::Counter {
+                    value: index as f64,
+                },
+            ))
+        })
+        .collect();
+    let stream = map_event_batch_stream(stream::iter(events.clone()), Some(batch));
+
+    let _ = sink.run(stream).await.unwrap();
+    assert_eq!(receiver.await, BatchStatus::Delivered);
+}
```

Comment:
The test itself seems fine, but from what I can tell, this is going to run for any ol' `cargo test` call, whereas this really needs to be an integration-level test, and ideally CI only.

Reply:
For running it as an integration test, you could look at the existing way we do this, which is with a feature flag that looks like:

```rust
#![cfg(feature = "aws-cloudwatch-logs-integration-tests")]
```

If you search for that specific feature flag, you can see how we organize them in Cargo.toml and what a new one would need to look like.

As for the fact that this should really only run in CI, that part I'm a little less sure on.  I'm not sure that we have a feature flag that we only enable when in CI.

@jszwedko Thoughts?

Reply:
👍 test looks good.

I would suggest we just add a new `datadog-integration-tests` feature that, typically, only CI would run, but could also be run locally via `make test-integration-datadog` if you export the appropriate env var. I might just call the env var `TEST_DATADOG_API_KEY` here and map `CI_TEST_DATADOG_API_KEY` to that in CI, but that's just a nit.

Reply:
Yeah, I mentioned all that in the PR description 😉 Thank you for those advices, I'll add this.